### ISSUE TITLE
Can't reuse AMQPMessage object with new properties

### ISSUE THREAD

**Issue by p-golovin** [php-amqplib/php-amqplib#576]
If I reuse AMQPMessage for performance optimisation as recommended in guide (https://github.com/php-amqplib/php-amqplib#optimized-message-publishing), I can't change message properties via set method (priority for example), because of this code
```
        if (!empty($this->serialized_properties)) {
            return $this->serialized_properties;
        }
```
in \PhpAmqpLib\Wire\GenericContent::serialize_properties

May be it will be a good idea set  $this->serialized_properties to null, when any property is changed with set method and new value don't equal to old value?

**Comment by p-golovin** [comment]
https://github.com/php-amqplib/php-amqplib/pull/577

**Comment by lukebakken** [comment]
Resolved by #577
